test/mlabns/mlabns.cpp

```diff
@@ -35,12 +38,201 @@ TEST_CASE("Query can pass the settings to the dns level") {
     std::string tool = "neubot";
 
     loop_with_initial_event([=]() {
-        mlabns::query(
-            tool, [](Error error, mlabns::Reply reply) {
-                REQUIRE(error);
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE("Make sure that an error is passed to callback with invalid "
+          "address_family settings") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ip4"; // Invalid
+    settings["mlabns/metro"] = "trn";
+    settings["mlabns/policy"] = "random";
+    std::string tool = "neubot";
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE("Make sure that an error is passed to callback with invalid metro "
+          "settings") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ipv4";
+    settings["mlabns/metro"] = "trno"; // Invalid
+    settings["mlabns/policy"] = "random";
+    std::string tool = "neubot";
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE("Make sure that an error is passed to callback with invalid policy "
+          "settings") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ipv4";
+    settings["mlabns/metro"] = "trn";
+    settings["mlabns/policy"] = "antani"; // Invalid
+    std::string tool = "neubot";
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE("Make sure that an error is passed to callback with invalid tool "
+          "settings") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ipv4";
+    settings["mlabns/metro"] = "trn";
+    settings["mlabns/policy"] = "random";
+    std::string tool = "antani"; // Invalid
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+MK_DEFINE_ERR(0001, MockError, "mock_error 0001")
```

Comment:
Actually there is a `MockError` class defined in `<common/error.hpp>` that we should instead use here. I will apply this change as part of merging the pull request.